src/lib.rs

```diff
@@ -314,28 +320,28 @@ impl Game {
         self.screen.as_ptr()
     }
 
-    pub fn input(&mut self, input_code : u8) {
-
-        // the layout is as follows:
-        // bit 0 - void
-        // bit 1 - up-key pressed
-        // bit 2 - down-key pressed
-        // bit 3 - left-key pressed
-        // bit 4 - right-key pressed
-
-        if (input_code & 2) != 0 {
-            if let Some(new_block) = rotate_left(&self.grid, &self.block) {
-                self.block = new_block;
+    /// bit 0 - void
+    /// bit 1 - up-key pressed
+    /// bit 2 - down-key pressed
+    /// bit 3 - left-key pressed
+    /// bit 4 - right-key pressed
+    pub fn input(&mut self, input_code: u8) {
+        match input_code {
+            2 => {
+                if let Some(new_block) = rotate_left(&self.grid, &self.block) {
+                    self.block = new_block;
+                }
             }
-        }
-        if (input_code & 4) != 0 {
-            drop_down(&mut self.grid, &mut self.block);
-        }
-        if (input_code & 8) != 0 {
-            step_left(&self.grid, &mut self.block);
-        }
-        if (input_code & 16) != 0 {
-            step_right(&self.grid, &mut self.block);
+            4 => {
```

Comment:
the method is supposed to work if the user presses multiple keys, this is not given anymore.

Reply:
Good point!